mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -5238,6 +5238,11 @@ static LogicalResult verifyVectorShapeCast(Operation *op,
     if (!isValidShapeCast(resultShape, sourceShape))
       return op->emitOpError("invalid shape cast");
   }
+
+  // Check that (non-)scalability is preserved
+  if (sourceVectorType.isScalable() != resultVectorType.isScalable())
+    return op->emitOpError("non-matching scalability flags");
```

Comment:
I think this would still allow `vector.shape_cast %foo : vector<[2]x[2]xf32> to vector<2x[2]xf32>`

Reply:
That's now updated :)